File: 01-OOP-in-C/src/person.c
#include "person.h"
#include <stdio.h>

void person_print(Person *this) {
  printf("Person\n\tname: %s\n\tsurname:%s\n\tage: %d\n\n", this->name,
         this->surname, this->age);
}

Person person_create(char *name, char *surname, int age) {
  Person person;

  person.name = name;
  person.surname = surname;
  person.age = age;

  return person;
}

File: 03-Rust-Hello-World/src/person.rs
/// This struct represents a person with a name, surname, and age.
///
pub struct Person {
    name: String,
    surname: String,
    age: u32,
}

impl Person {
    pub fn create(name: String, surname: String, age: u32) -> Person {
        return Person {
            name: name,
            surname: surname,
            age,
        };
    }

    pub fn print(&self) {
        println!(
            "Name: {}, Surname: {}, Age: {}",
            self.name, self.surname, self.age
        );
    }
}
